modules/javafx.graphics/src/main/java/com/sun/javafx/application/preferences/PreferenceProperties.java

```diff
@@ -0,0 +1,240 @@
+                }
+            }
+        }
+
+        fireValueChangedEvent();
+    }
+
+    private void updateColorProperty(ColorProperty property, Object value) {
+        if (value instanceof Color color) {
+            property.setValue(color);
+        } else {
+            if (value != null) {
+                Logging.getJavaFXLogger().warning(
+                    "Unexpected value of " + property.getName() + " platform preference, " +
+                    "using default value instead (expected = " + Color.class.getName() +
+                    ", actual = " + value.getClass().getName() + ")");
+            }
+
+            // Setting a ColorProperty to 'null' restores its platform value or default value.
+            property.setValue(null);
+        }
+    }
+
+    private void fireValueChangedEvent() {
+        for (ColorProperty colorProperty : allColors) {
+            colorProperty.fireValueChangedEvent();
+        }
+    }
+
+    /**
+     * ColorProperty implements a deferred notification mechanism, where change notifications
+     * are only fired after changes of all color properties have been applied.
+     * This ensures that observers will never see a transient state where two color properties
+     * are inconsistent (for example, both foreground and background could be the same color
+     * when going from light to dark mode).
+     */
+    private final class ColorProperty extends ReadOnlyObjectPropertyBase<Color> {
+        private final String name;
+        private final Color defaultValue;
+        private Color overrideValue;
+        private Color platformValue;
+        private Color effectiveValue;
+        private Color lastEffectiveValue;
+
+        ColorProperty(String name, Color initialValue) {
+            this.name = name;
+            this.defaultValue = initialValue;
+            this.platformValue = initialValue;
+            this.effectiveValue = initialValue;
+            this.lastEffectiveValue = initialValue;
+        }
+
+        @Override
+        public Object getBean() {
+            return bean;
+        }
+
+        @Override
+        public String getName() {
+            return name;
+        }
+
+        @Override
+        public Color get() {
+            return effectiveValue;
+        }
+
+        /**
+         * Only called by {@link #updateColorProperty}, this method doesn't fire a change notification.
+         * Change notifications are fired after the new values of all color properties have been set.
+         */
+        public void setValue(Color value) {
+            this.platformValue = value;
+            updateEffectiveValue();
+        }
+
+        public void setValueOverride(Color value) {
+            this.overrideValue = value;
+            updateEffectiveValue();
+            fireValueChangedEvent();
+        }
+
+        private void updateEffectiveValue() {
+            // Choose the first non-null value in this order: overrideValue, platformValue, defaultValue.
```

Comment:
are null default values allowed?

Reply:
I can't think of a good reason why we would have preferences with a `null` default value. The small set of convenience properties is meant as a baseline that all JavaFX developers can rely on.